embassy-mcxa/src/wwdt.rs

```diff
@@ -0,0 +1,220 @@
+//! Windowed Watchdog Timer (WWDT) driver for MCXA microcontrollers.
+//!
+//! The WWDT is a hardware timer that can reset the system or generate an interrupt if the software fails to
+//! periodically "feed" the watchdog within a specified time window. This helps detect
+//! and recover from software failures or system hangs.
+//! The FRO12M provides a 1 MHz clock (clk_1m) used as WWDT0 independant clock source. This clock is / 4 by an internal fixed divider.
+//!
+
+use crate::interrupt::typelevel;
+use crate::interrupt::typelevel::Handler;
+use crate::pac;
+use crate::peripherals::WWDT0;
+use embassy_hal_internal::interrupt::InterruptExt;
+use embassy_hal_internal::Peri;
+use embassy_time::Duration;
+
+/// Watchdog peripheral
+pub struct Watchdog<'d> {
+    _peri: Peri<'d, WWDT0>,
+    load_value: u32,
+    frequency: u32,
+    // The register block of the WWDT instance
+    info: &'static pac::wwdt0::RegisterBlock,
+}
+
+impl<'d> Watchdog<'d> {
+    /// Create a new WWDT instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `_peri` - The WWDT0 peripheral instance
+    /// * `_irq` - Interrupt binding for WWDT0
+    pub fn new(
+        _peri: Peri<'d, WWDT0>,
+        _irq: impl crate::interrupt::typelevel::Binding<typelevel::WWDT0, InterruptHandler> + 'd,
+    ) -> Self {
+        let _frequency = crate::clocks::with_clocks(|clocks| {
+            // Ensure clk_1m is active at the required power level
+            clocks.ensure_clk_1m_active(&crate::clocks::PoweredClock::NormalEnabledDeepSleepDisabled)
+        })
+        .expect("Clocks not initialized")
+        .expect("clk_1m not enabled or not at required power level");
+
+        // Enable WATCHDOG clock by writing to mrcc register
+        // Can't use enable_and_reset API here because WWDT doesn't have a reset signal.
+        let mrcc = unsafe { pac::Mrcc0::steal() };
+        mrcc.mrcc_glb_cc0().modify(|_, w| w.wwdt0().enabled());
+
+        crate::pac::Interrupt::WWDT0.unpend();
+        unsafe {
+            crate::pac::Interrupt::WWDT0.enable();
+        }
+        Self {
+            _peri,
+            load_value: 0,
+            frequency: _frequency / 4,
+            info: unsafe { &*pac::Wwdt0::ptr() },
+        }
+    }
+
+    // Start the watchdog timer with the specified timeout period.
+    ///
+    /// # Arguments
+    ///
+    /// * `period` - The timeout period after which the watchdog will trigger
+    /// * `warning` - Optional warning interrupt period before timeout. If `Some`, an interrupt
+    ///               will be generated at this time before the actual timeout. If `None`, the
+    ///               watchdog will directly reset the system on timeout.
+    ///
+    pub fn start(&mut self, period: Duration, warning: Option<Duration>) {
```

Comment:
Another way to do this would be to pass both `period` and `warning` as part of a `Config` struct to the constructor. Then pre-program the watchdog with everything you need, during construction and leave start with the single responsibility of starting it, that is: `enable(); feed();`